fastchat/model/model_adapter.py

```diff
@@ -99,16 +99,17 @@ def register_model_adapter(cls):
 @cache
 def get_model_adapter(model_path: str) -> BaseModelAdapter:
     """Get a model adapter for a model_path."""
-    model_path_basename = os.path.basename(os.path.normpath(model_path))
-
-    # Try the basename of model_path at first
+    # Try the fullname of model_path at first.  Some models (namely Peft)
```

Comment:
I'm pretty sure this will break *something* else but i'm not sure what to do.

With the way lookup happens at head (looking up via basename) we get the wrong results when loading a peft model associated with an organization on HuggingFace.  This happens because:
  1. The peft model config can't be found locally nor remotely (the basename repo won't exist).
  2. We then match whatever base model seems suitable (in my test case this is Pythia or MPT given the model names).
  3. We later on try to load that base model and it fails given the full repo path since that points to a Peft repo.

So, in order to handle remote peft models correctly, we have to check the full path first.  However, I wasn't tracking what issue this ordering was meant to handle so I'd concerned this introduces yet another bug.